src/Soil/energy_hydrology.jl

```diff
@@ -0,0 +1,143 @@
+struct HeatParameters{FT <: AbstractFloat}
+    "The thermal conductivity"
+    κ::FT
+    "The volumetric heat capacity"
+    ρc_s::FT
+end
+
+"""
+    SoilEnergyHydrology
+
+A model for simulating the flow of water and heat 
+in a porous medium by solving the Richardson-Richards equation
+and the heat equation, including terms for phase change.
+
+$(DocStringExtensions.FIELDS)
+"""
+struct SoilEnergyHydrology{FT, PSR, PSH, D, C, BCR, BCH, S} <:
+       AbstractSoilModel{FT}
+    "the parameter set for RR equation"
+    rre_param_set::PSR
```

Comment:
might be nicer to have params.rre, params.heat? same goes for boundary conditions.